[EDU-5536] Videoteca Player integration guide (#1386)
aziontech/docs: src/content/docs/en/pages/guides/marketplace/integrations/videoteca-player.mdx

---
title: How to install the Videoteca Player integration through Azion Marketplace
description: >-
  Videoteca Player provides a quick way to host your own Videofront's Videoteca player.
meta_tags: 'videoteca, video player, edge computing, integration'
namespace: docs_use_case_videoteca_player

> permalink: /documentation/products/guides/videoteca-player/

Author: ```suggestion
permalink: /documentation/products/guides/videofront-player/
```

---

import Tabs from '~/components/tabs/Tabs'
import InterfaceNote from '~/includes/snippets/InterfaceNote/en/snippet.mdx'

<InterfaceNote />

> The **Videoteca Player** integration offers a fast and efficient solution for hosting [Videofront's Videoteca](https://videofront.com.br/videoteca) player using the iframe embedding method. With this integration, you can seamlessly load videos by simply passing a query string argument, `video-id`, to specify the content to be displayed.

Reviewer: ```suggestion
The **Videoteca Player** integration offers a fast and efficient solution for hosting [Videofront's Videoteca](https://videofront.com.br/videoteca) player using the iframe embedding method. With this integration, you can load videos by simply passing the `video-id` query string argument to specify the content to be displayed.
```

---

## Requirements

To set up this integration, you need to provide a **Username** and an **API Token** from **Videoteca**. To obtain these credentials, you need to have an account on the [Videoteca website](https://videofront.com.br/videoteca).

---

## Getting the integration

> To install this integration provided by Azion's Marketplace, you have to:

Reviewer: ```suggestion
To install this integration provided by Azion Marketplace, you have to:
```

1. Access [Azion Console](/en/documentation/products/guides/how-to-access-azion-console/) > **Marketplace**.
2. On the Marketplace homepage, select the integration's card.
3. Once the integration's page opens, click the **Install** button, at the bottom-right corner of the page.

You'll see a message indicating that your integration was successfully installed.
</Fragment>

<Fragment slot="panel.rtm">
1. Access [Real-Time Manager (RTM)](https://manager.azion.com/) > **Marketplace**.
2. On the Marketplace homepage, select the integration's card.
3. Once the integration's page opens, click the **Get It Now** button, at the bottom-right corner of the page.

You'll see a message indicating that your integration was successfully installed.
</Fragment>

</Tabs>

:::tip
You can search any integration by browsing through the cards, using the filters, or typing a keyword in the search bar. 
:::

---

> ## Getting the keys

Reviewer: What do you think about including this info as Requirements, in the first h2?

To configure this integration, you have to provide a **Videoteca** Username and API Token. To get these credentials, you must have an account at the [Videoteca site](https://videofront.com.br/videoteca).

---

## Setting up the integration

### Instantiating the edge function

To create an instance of the integration's function:

<Tabs client:visible>
    <Fragment slot="tab.console">Console</Fragment>
    <Fragment slot="tab.rtm">Real-Time Manager</Fragment>

<Fragment slot="panel.console">

> 1. On the **Products Menu**, navigate to **Build** > **Edge Application**.

Reviewer: ```suggestion
1. On the **Products menu**, navigate to **Build** > **Edge Application**.
```

2. Select the application from the list to which you want to add the integration or [create a new edge application](/en/documentation/products/build/edge-application/first-steps/).
3. Select the **Functions Instances** tab.
4. Click the **+ Function Instance** button.
5. Give a name to your instance.
6. On the dropdown menu, select the *Videoteca Player* function.
  - This action will load the **Arguments** tab, where you can add the parameters to execute your application.

> 7. In the **Arguments** tab, pass the three parameters:

Reviewer: ```suggestion
7. In the **Arguments** tab, pass these parameters:
```

```json
{
"videoteca_user": "<your_username>", // Replace with your Videoteca username
"videoteca_token": "<your_token>", // Replace with your Videoteca API token
"custom_html": "<your_html>" // Replace with your custom html
}
```

Where:

| Variable | Required | Description |
| ---- | ----- | ----- |

> | `videoteca_user` | Yes | The name of the account from where the function will retrieve the videos. |
> | `videoteca_token` | Yes | The token to access the Videoteca's API. |
> | `custom_html` | No | Custom HTML in the page delivered by the function. It must include the tag `<!-- videoteca_iframe -->` in the HTML, so the function will replace it with the actual iframe of the video. |

Reviewer: ```suggestion
| `videoteca_user` | Yes | The name of the account from where the function will retrieve the videos |
| `videoteca_token` | Yes | The token to access the Videoteca's API |
| `custom_html` | No | Custom HTML in the page delivered by the function. It must include the tag `<!-- videoteca_iframe -->` in the HTML, so the function will replace it with the actual iframe of the video |
```

Reviewer: style suggestion

> | ---- | ----- | ----- |
> | `videoteca_user` | Yes | The name of the account from which the function will retrieve the videos. |
> | `videoteca_token` | Yes | The token to access the Videoteca's API. |
> | `custom_html` | No | Custom HTML for the page delivered by the function. It must include the tag `<!-- videoteca_iframe -->` in the HTML, as the function will replace it with the actual iframe of the video. |

Reviewer: ```suggestion
| ---- | ----- | ----- |
| `videoteca_user` | Yes | The name of the account from which the function will retrieve the videos |
| `videoteca_token` | Yes | The token to access the Videoteca's API |
| `custom_html` | No | Custom HTML for the page delivered by the function. It must include the tag `<!-- videoteca_iframe -->` in the HTML, as the function will replace it with the actual iframe of the video |
```

:::tip
You can get your token at: `https://<YOUR-USER>.videotecaead.com.br/Apidoc` replacing `<YOUR-USER>` with your actual Videoteca username.
:::

> 6. Click the **Save** button to save your configuration.

Reviewer: ```suggestion
8. Click the **Save** button to save your configuration.
```

Done. Now your Videoteca Player instance is saved.
</Fragment>

<Fragment slot="panel.rtm">

> 1. On the **Products Menu**, navigate to **BUILD** > **Edge Application**.

Reviewer: ```suggestion
1. On the **Products menu**, navigate to **BUILD** > **Edge Application**.
```

2. Select the application from the list to which you want to add the integration or [create a new edge application](/en/documentation/products/build/edge-application/first-steps/).

> 1. Select the **Functions** tab.
> 2. Click the **Add Function** button.
> 3. Give a name to your instance.
> 4. On the dropdown menu, select the *Videoteca Player* function.
> - This action will load the function, showing a form with the function's source-code and, just above it, two tabs: **Code** and **Args**. By clicking on the **Code** tab, you'll be able to navigate through the source-code, but won't be able to change it.
> 5. In the **Args** tab, you'll pass the two parameters:

Reviewer: ```suggestion
3. Select the **Functions** tab.
4. Click the **Add Function** button.
5. Give a name to your instance.
6. On the dropdown menu, select the *Videoteca Player* function.
- This action will load the function, showing a form with the function's source-code and, just above it, two tabs: **Code** and **Args**. By clicking on the **Code** tab, you'll be able to navigate through the source-code, but won't be able to change it.
7. In the **Args** tab, you'll pass the two parameters:
```

> 5. In the **Args** tab, you'll pass the two parameters:

Author: ```suggestion
5. In the **Args** tab, you'll pass these parameters:
```

```json
{
"videoteca_user": "<your_username>", // Replace with your Videoteca username
"videoteca_token": "<your_token>", // Replace with your Videoteca API token
"custom_html": "<your_html>" // Replace with your custom html
}
```

Where:

> | Variable | Required | Description |

Reviewer: apply suggestions in Console table, if ok for you.

| ---- | ----- | ----- |

> | `videoteca_user` | Yes | The name of the account from which the function will retrieve the videos. |
> | `videoteca_token` | Yes | The token to access the Videoteca's API. |
> | `custom_html` | No | Custom HTML for the page delivered by the function. It must include the tag `<!-- videoteca_iframe -->` in the HTML, as the function will replace it with the actual iframe of the video. |

Reviewer: ```suggestion
| `videoteca_user` | Yes | The name of the account from which the function will retrieve the videos |
| `videoteca_token` | Yes | The token to access the Videoteca's API |
| `custom_html` | No | Custom HTML for the page delivered by the function. It must include the tag `<!-- videoteca_iframe -->` in the HTML, as the function will replace it with the actual iframe of the video |
```

:::tip
You can get your token at: `https://<YOUR-USER>.videotecaead.com.br/Apidoc` replacing `<YOUR-USER>` with your actual Videoteca username.
:::

> 6. Click the **Save** button to save your configuration.

Reviewer: ```suggestion
8. Click the **Save** button to save your configuration.
```

Done. Now your Videoteca Player instance is saved.
</Fragment>

</Tabs>

### Setting up the Edge Application Rules Engine

To finish, you have to set up the **Rules Engine** to configure the *behavior* and the *criteria* to run the function.

Still on the **Edge Application** page:

<Tabs client:visible>
    <Fragment slot="tab.console">Console</Fragment>
    <Fragment slot="tab.rtm">Real-Time Manager</Fragment>

<Fragment slot="panel.console">
1. Select the **Rules Engine** tab.
2. Click the **+ Rule Engine** button.
3. Give a name to the rule.
4. In **Criteria**, define the conditions for running the function.

> 5. Below, select a *behavior* for when the *criteria* are met. In this case, it'll be **Run Function**. 

Reviewer: ```suggestion
5. Below, select the **Run Function** behavior. 
```

    - Select the adequate function according to the name you gave it in the instantiate step.
6. Click the **Save** button.

Done. Now the integration will run according to the criteria you set.
</Fragment>

<Fragment slot="panel.rtm">
1. Select the **Rules Engine** tab.
2. Click the **New Rule** button.
3. Give a name to the rule.
4. In **Criteria**, define the conditions for running the function.

> 5. Below, select a *behavior* for when the *criteria* are met. In this case, it'll be **Run Function**.

Reviewer: ```suggestion
5. Below, select the **Run Function** behavior. .
```
